proof_range.go

```diff
@@ -472,6 +518,7 @@ func (t *ImmutableTree) getRangeProof(keyStart, keyEnd []byte, limit int) (proof
 				h := sha256.Sum256(node.value)
 				leaves = append(leaves, ProofLeafNode{
 					Key:       node.key,
+					Value:     node.value,
```

Comment:
Why do we need the values for the leaf nodes? I believe you need the values for the inner nodes for rebalancing, rather than leaf nodes.

Reply:
Sorry I misinterpreted it to mean values of the node, not just the `valueHash`. I'll add the values of the `leftNode` and `rightNode` in `ProofInnerNode` in addition to the already existing `leftHash` and `rightHash`

Reply:
Added it: https://github.com/celestiaorg/iavl/pull/3/commits/7b89c40f83f913edf7551860db077b2dcafc566e

Reply:
To clarify I believe we need the preimage of sibling inner nodes, not their hashes, to get their "size" field to enable set() to do rebalancing. This is why doing the full test I suggested below will be important - as I'm not sure sets will work right now as they don't have the data needed to do rebalancing.